app/actions/LNActions.js

```diff
@@ -0,0 +1,612 @@
+import * as ln from "wallet/ln";
+import * as sel from "selectors";
+import * as wallet from "wallet";
+import { ipcRenderer } from "electron";
+import { getWalletCfg } from "../config";
+import { getWalletPath } from "main_dev/paths";
+
+export const CLOSETYPE_COOPERATIVE_CLOSE = 0;
+export const CLOSETYPE_LOCAL_FORCE_CLOSE = 1;
+export const CLOSETYPE_REMOTE_FORCE_CLOSE = 2;
+export const CLOSETYPE_BREACH_CLOSE = 3;
+export const CLOSETYPE_FUNDING_CANCELED = 4;
+export const CLOSETYPE_ABANDONED = 5;
+
+export const LNWALLET_STARTDCRLND_ATTEMPT = "LNWALLET_STARTDCRLND_ATTEMPT";
+export const LNWALLET_STARTDCRLND_FAILED = "LNWALLET_STARTDCRLND_FAILED";
+export const LNWALLET_STARTDCRLND_SUCCESS = "LNWALLET_STARTDCRLND_SUCCESS";
+
+export const startDcrlnd = (passphrase, autopilotEnabled, walletAccount) => async (dispatch, getState) => {
+
+  dispatch({ type: LNWALLET_STARTDCRLND_ATTEMPT });
+
+  const { grpc: { port } } = getState();
+  const { daemon: { credentials, appData, walletName } } = getState();
+  const isTestnet = sel.isTestNet(getState());
+  const walletPath = getWalletPath(isTestnet, walletName);
+  const walletPort = port;
+  let rpcCreds = {};
+
+  const cfg = getWalletCfg(isTestnet, walletName);
+  const lnCfg = dispatch(getLNWalletConfig());
+  const cliOptions = ipcRenderer.sendSync("get-cli-options");
+
+  if (typeof walletAccount != "number") {
+    walletAccount = lnCfg.account;
+  }
+
+  if (cliOptions.rpcPresent) {
+    rpcCreds.rpc_user = cliOptions.rpcUser;
+    rpcCreds.rpc_pass = cliOptions.rpcPass;
+    rpcCreds.rpc_cert = cliOptions.rpcCert;
+    rpcCreds.rpc_host = cliOptions.rpcHost;
+    rpcCreds.rpc_port = cliOptions.rpcPort;
+  } else if (credentials) {
+    rpcCreds.rpc_user = credentials.rpc_user;
+    rpcCreds.rpc_cert = credentials.rpc_cert;
+    rpcCreds.rpc_pass = credentials.rpc_pass;
+    rpcCreds.rpc_host = credentials.rpc_host;
+    rpcCreds.rpc_port = credentials.rpc_port;
+  } else if (appData) {
+    rpcCreds.rpc_user = cfg.get("rpc_user");
+    rpcCreds.rpc_pass = cfg.get("rpc_pass");
+    rpcCreds.rpc_cert = `${appData}/rpc.cert`;
+    rpcCreds.rpc_host = cfg.get("rpc_host");
+    rpcCreds.rpc_port = cfg.get("rpc_port");
+  } else {
+    rpcCreds.rpc_user = cfg.get("rpc_user");
+    rpcCreds.rpc_pass = cfg.get("rpc_pass");
+    rpcCreds.rpc_host = cfg.get("rpc_host");
+    rpcCreds.rpc_port = cfg.get("rpc_port");
+  }
+
+  try {
+    const res = ipcRenderer.sendSync("start-dcrlnd", walletAccount, walletPort,
+      rpcCreds, walletPath, isTestnet, autopilotEnabled);
+
+    try {
```

Comment:
there is 2 try catch here

Reply:
This is needed because the first check might fail but with an error that still allows us to continue with the connection attempt.

Specifically, error code 12 means the node is up but hasn't been unlocked yet, so we can just try to unlock it.